# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineIC.cpp
# Commit: 7bd0784e7a6f
# Full Hash: 7bd0784e7a6f3702152bdca395b2305f16ca0479
# Author: Chris Fallin <cfallin@mozilla.com>
# Date: 2019-12-03 09:48:30
# Regressor Bug: 1598347
# File Overlap Count: 1
# Description:
#   Bug 1598347, part 2: pass "inner singleton" to NEWOBJECT group logic. r=djvj,iain
#   
#   This change passes through the "inner singleton" status of a particular
#   object literal to the group-assignment logic for JSOP_NEWOBJECT by
#   instead using a variant opcode, JSOP_NEWOBJECT_WITHGROUP.  "Inner
# ==============================================================================

diff -r 6da3c648995f -r 7bd0784e7a6f js/src/jit/BaselineIC.cpp
--- a/js/src/jit/BaselineIC.cpp	Wed Nov 27 22:48:40 2019 +0000
+++ b/js/src/jit/BaselineIC.cpp	Wed Nov 27 22:49:33 2019 +0000
@@ -324,6 +324,7 @@
         break;
       }
       case JSOP_NEWOBJECT:
+      case JSOP_NEWOBJECT_WITHGROUP:
       case JSOP_NEWINIT: {
         ICStub* stub = alloc.newStub<ICNewObject_Fallback>(Kind::NewObject);
         if (!addIC(loc, stub)) {